edgelet/edgelet-http/src/util/hyperwrap.rs

```diff
@@ -46,6 +49,21 @@ impl Config {
     }
 }
 
+fn uri_to_proxy(uri: Uri) -> Result<Proxy, Error> {
+    let cloned = uri.clone();
+    let url = Url::parse(&uri.to_string())?;
```

Comment:
We have to reparse the url because `hyper::Uri` doesn't give access to the username/password. The `url` crate does provide access.